File: arrays-Hashing/leetCode_1.py
"""
Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.

You can return the answer in any order.
"""
class Solution:
    def twoSum(self, nums, target):
        hashMap = {}
        for index, val in enumerate(nums):
            difference = target - val
            if difference in hashMap:
                return [hashMap[difference],index]
            hashMap[val] = index
        return []

s = Solution()
print(s.twoSum([3,2,4], 6))

File: arrays-Hashing/leetCode_1.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class leetCode_1 {
    public int[] twoSum(int[] nums, int target){
        Map<Integer,Integer> numMap = new HashMap<>();
        int n = nums.length;
        for (int i=0; i<n;i++){
            int complement = target - nums[i];
            if (numMap.containsKey(complement)){
                return new int[] {numMap.get(complement), i};
            }
            numMap.put(nums[i],i);
        }
        return new int[]{};
    }

    public static void main(String[] args) {
        leetCode_1 solution = new leetCode_1();
        int []nums = {2,7,9,11};
        int target = 9;
        int[] result = solution.twoSum(nums, target);
        System.out.println("The indexes are: "+Arrays.toString(result));
    }
}
